python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt_tests/dlt_tests/test_dlt.py

```diff
@@ -205,3 +207,27 @@ def example_pipeline_assets(
     assert not res.success
     asset_materizations = res.get_asset_materialization_events()
     assert len(asset_materizations) == 0
+
+
+def test_partitioned_materialization(dlt_pipeline: Pipeline) -> None:
+    @dlt_assets(
+        dlt_source=pipeline(),
+        dlt_pipeline=dlt_pipeline,
+        partitions_def=MonthlyPartitionsDefinition(start_date="2022-08-09"),
+    )
+    def example_pipeline_assets(
+        context: AssetExecutionContext, dlt_pipeline_resource: DagsterDltResource
+    ):
+        yield from dlt_pipeline_resource.run(context=context)
+
+    async def run_partition(year: str):
+        return materialize(
+            [example_pipeline_assets],
+            resources={"dlt_pipeline_resource": DagsterDltResource()},
+            partition_key=year,
+        )
```

Comment:
Hi @edsoncezar16 - thanks for helping on this.

I would be curious to learn more about how you are using this feature. I would have thought that for partitioning to be meaningful in the context of _dlt_, we would need to have access to the partition parameters (start, end) from the _dlt_ pipeline itself. That way we can do conditional / incremental loading.

We update the `pipeline_name`, but do we also need to pass the partitions to the `run` method?

```
load_info = dlt_pipeline.run(dlt_source, **kwargs)
```

Reply:
Hi, I have updated the test to also show an example of using the partition key to modify the dlt resource at run time. Notice that this is only possible because we modified the DagsterDltResource.run method, so that it gives precedence to explicit args even for Asset Executions (and not only for Op executions).

Reply:
Ok, that makes sense to me.

I do hope that unwanted behavior might not arise by having a possibly different invocation of the pipeline in the `dlt_assets` decorate, and the one passed to the `run` method. I could imagine a resource not being used based on some condition, and then we have deviations between the two.

But for now, I can't think of any other way to accomplish this, and it doesn't impact existing jobs which is good.

FYI, @edsoncezar16 there are some small merge conflicts with the metadata PR that need to be resolved.

Reply:
Sure, I will fix those conflicts in a moment. TYVM!